### PR TITLE
Kajam 11.12.0 plat 4645 new xslt

### PR DESCRIPTION
**Pull request by noam-arad**

### CODE REVIEW

File: deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+$c->add(DistributionProfilePeer::PROVIDER_TYPE, TvinciDistributionPlugin::getDistributionProviderTypeCoreValue(TvinciDistributionProviderType::TVINCI));
+$distributionProfiles = DistributionProfilePeer::doSelect($c, $con);
+
+foreach ($distributionProfiles as $distributionProfile)
+{
+	/**
+	 * @var TvinciDistributionProfile $distributionProfile
+	 */
+	$tags = array();
+	$ismFileName = $distributionProfile->getFromCustomData('ismFileName', null, '');
```
Comment (EladCohenKaltura) [inline deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php]: Seems like a function would shorten this code significantly

File: deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	$ismPpvmodule = $distributionProfile->getFromCustomData('ismPpvModule', null, '');
+	if ( $ismFileName != '' || $ismPpvmodule != '' )
```
Comment (EladCohenKaltura) [inline deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php]: since there was no delete to field there are file name and ppv module with space as empty , trim before comparing to be on the safe side

File: deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+	$widevineFileName = $distributionProfile->getFromCustomData('widevineFileName', null, '');
+	$widevinePpvmodule = $distributionProfile->getFromCustomData('widevinePpvModule', null, '');
+	if ( $widevineFileName != '' || $widevinePpvmodule != '' )
+	{
+		$tag = createTvinciTag("widevine", $widevineFileName, $widevinePpvmodule);
+		$tags[] = $tag;
+	}
+
+	$widevine_mbrFileName = $distributionProfile->getFromCustomData('widevineMbrFileName', null, '');
```
Comment (EladCohenKaltura) [inline deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php]: widevine_mbrFileName Vs. widevineMbrFileName - I think widevineMbrFileName is better :)

File: deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+	switch ($tagname)
+	{
+		case 'ism':
+			$newTag->setFormat('sl');
+			$newTag->setProtocol('http');
+			$newTag->setExtension('ism');
+			break;
+		case 'ipadnew':
+			$newTag->setFormat('applehttp');
```
Comment (EladCohenKaltura) [inline deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php]: no need for code in case - combine the ipadnew and iphonenew to same case

File: deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+			$newTag->setProtocol('http');
+			$newTag->setExtension('a4m');
+			break;
+		case 'dash':
+			$newTag->setFormat('mpegdash');
+			$newTag->setProtocol('http');
+			$newTag->setExtension('mpd');
+			break;
+		case 'widevine':
+			$newTag->setFormat('url');
```
Comment (EladCohenKaltura) [inline deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php]: same here

File: infra/content/kXml.class.php [hunk 6]
```diff
@@ -27,6 +27,12 @@ public function errorHandler($errNo, $errStr, $errFile, $errLine)
 	}
 }
 
+function xml_load_for_xslt($xmlStr)
```
Comment (EladCohenKaltura) [inline infra/content/kXml.class.php]: Did you try and manipulate http://www.w3schools.com/xsl/func_document.asp to work ? instead of calling PHP

File: plugins/content_distribution/providers/tvinci/lib/admin/TvinciProfileConfiguration.php [hunk 7]
```diff
@@ -110,13 +118,11 @@ protected function tvinciElements()
...
 
 		// tag specific configuration
-		$this->addTagItems("ism");
-		$this->addTagItems("ipadnew");
-		$this->addTagItems("iphonenew");
-		$this->addTagItems("mbr");
-		$this->addTagItems("dash");
-		$this->addTagItems("widevine");
-		$this->addTagItems("widevine_mbr");
+		$tvinciDistributionTagsSubForm = new Zend_Form_SubForm(array('DisableLoadDefaultDecorators' => true));
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/admin/TvinciProfileConfiguration.php]: This happens twice once in the populateFromObject function and once here 
is this a different load path and if so - create a func that does this

File: plugins/content_distribution/providers/tvinci/lib/feed/TvinciDistributionFeedHelper.php [hunk 8]
```diff
@@ -57,34 +57,88 @@ private function createArgumentsForXSLT()
...
+		$tagsDoc = new DOMDocument();
+		$tagsConfiguration = $tagsDoc->createElement('tagsconfiguration');
+		foreach($this->distributionProfile->tags as $tag)
+		{
+			/**
+			 * @var KalturaTvinciDistributionTag $tag
+			 */
+			$tagXML = $tagsDoc->createElement('tag');
+
+			$tagname = $tagsDoc->createElement('tagname');
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/feed/TvinciDistributionFeedHelper.php]: create a function "append tag element" or something like that for all these similar calls

File: plugins/content_distribution/providers/tvinci/lib/feed/TvinciDistributionFeedHelper.php [hunk 9]
```diff
@@ -57,34 +57,88 @@ private function createArgumentsForXSLT()
...
+			}
+			if ($tag->tagname == 'widevineMbrFileName')
+			{
+				$arguments["widevineMbrPpvModule"] = $tag->ppvmodule;
+				$arguments["widevineMbrTypeName"] = $tag->filename;
+			}
+		}
+		$tagsDoc->appendChild($tagsConfiguration);
+		$arguments['tagsparam'] = $tagsDoc->saveXML();
+		KalturaLog::err("@@NA arguments [".print_r($arguments,true)."]");
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/feed/TvinciDistributionFeedHelper.php]: Please remove

File: plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt [hunk 10]
```diff
@@ -452,31 +457,28 @@
                     </xsl:for-each>
                 </xsl:variable>
 
-                <xsl:variable name="suffix">
-                    <xsl:call-template name="suffix-matching-tag">
-                        <xsl:with-param name="tag" select="$tag"/>
-                    </xsl:call-template>
+                <xsl:variable name="nameSuffix">
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt]: please move this to be a function

File: plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt [hunk 11]
```diff
@@ -452,31 +457,28 @@
...
-                <xsl:variable name="ppvModule">
-                    <xsl:call-template name="ppvModule-matching-tag">
-                        <xsl:with-param name="tag" select="$tag"/>
-                    </xsl:call-template>
+                    <xsl:value-of select="concat($tag/typename, $nameSuffix)" />
                 </xsl:variable>
 
                 <xsl:call-template name="create-file-element">
-                    <xsl:with-param name="cdnCode" select="concat($playManifestPrefix, $relevantEntryId, $suffix)"/>
+                    <xsl:with-param name="cdnCode" select="concat($playManifestPrefix, $relevantEntryId, '/format/',$tag/format, '/tags/', $tag/tagname, '/protocol/',$tag/protocol,'/f/a.',$tag/extension)"/>
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt]: I would extract the suffix into a function as well - since this can be a place where the PS makes their changes it will be more usable if has the scope of a function

File: plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt [hunk 12]
```diff
@@ -579,104 +581,5 @@
...
-                        <xsl:value-of select="concat(' ',$childNumber)"/>
-                    </xsl:if>
-                </xsl:variable>
-                <xsl:value-of select="concat($trailerSuffix, $trailerIdxSuffix)"/>
-            </xsl:if>
-        </xsl:variable>
-        <xsl:value-of select="concat($configuredName, $nameSuffix)"/>
-    </xsl:template>
 
 </xsl:stylesheet>
```
Comment (EladCohenKaltura) [inline plugins/content_distribution/providers/tvinci/lib/xml/tvinci_default.xslt]: Did you check the case when you put two 'same tag yet different' configurations and got two different files ?

File: release-notes.md [hunk 13]
```diff
@@ -1,3 +1,36 @@
...
+#### Deployment scripts####
+
+	php /opt/kaltura/app/deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php <partner_id>
+
+#### Known Issues & Limitations ####
+- None.
+
+
+## Update comment_was_added_to_entry email notification ##
+
```
Comment (EladCohenKaltura) [inline release-notes.md]: How is this related ?

File: release-notes.md [hunk 14]
```diff
@@ -1,3 +1,36 @@
...
+ - Issue Type: Story
+ - Issue ID: PLAT-4617
+
+#### Configuration ####
+- None.
+
+#### Deployment scripts####
+
+	php /opt/kaltura/app/deployment/updates/scripts/2016_04_05_migrate_tvinci_distribution_tags_to_be_dynamic.php <partner_id>
```
Comment (EladCohenKaltura) [inline release-notes.md]: Since the script requires <partner id> I think you should mention which partners to run it on

**Comment by EladCohenKaltura** [comment]
Done!

File: plugins/content_distribution/providers/tvinci/lib/admin/TvinciProfileConfiguration.php (hunk 7 above)
Comment (noam-arad) [inline plugins/content_distribution/providers/tvinci/lib/admin/TvinciProfileConfiguration.php]: The one in the populate is in order to actually populate this, the one in tvinciElements is in order to place it correctly in the form (display wise).

File: release-notes.md (hunk 13 above)
Comment (noam-arad) [inline release-notes.md]: to avoid conflict with Gal's code

File: release-notes.md (hunk 14 above)
Comment (noam-arad) [inline release-notes.md]: This is general release notes and doesn't have partner specific information
